Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -124,6 +124,46 @@ public CsharpCompiler(ErrorReporter reporter)
           w.WriteLine("return a;");
         }
       }
+
+      //Generates casts for functions of those arities present in the program
+      var funcExtension = wr.NewNamedBlock("public static class FuncExtension");
```

Comment:
I think the common name for classes containing extension function is `<X>Extensions`, with the `s` at the end.